R/defStormsDataset.R

```diff
@@ -241,23 +318,294 @@ checkInputsdefStormsDataset <- function(filename, fields, basin, seasons, unitCo
+  # Filter by season
+  filter <- which(as.numeric(dataBaseFiltered[,fields["seasons"]]) >= seasons[1] & as.numeric(dataBaseFiltered[,fields["seasons"]]) <= seasons[2])
+  dataBaseFiltered <- dataBaseFiltered[filter,]
+  
+  # Filter by basin
+  if(!is.null((basin))){
+    filter <- which(dataBaseFiltered[,fields["basin"]] == basin)
+    dataBaseFiltered <- dataBaseFiltered[filter,]
+  }
+  
+  # Get dimensions
+  stormNames <- dataBaseFiltered[,fields["names"]] 
+  stormSeasons <- dataBaseFiltered[,fields["seasons"]]
+  sid <- paste0(stormNames, stormSeasons)
+
+  # Do the following code to replace table function
+  k <- 0
+  actualStorm <- sid[1]
+  countObs <- c()
+  names <- c()
+  for (i in 1:length(sid)){
+    if(sid[i] != actualStorm){
+      countObs <- c(countObs, k)
+      names <- c(names, actualStorm)
+      actualStorm <- sid[i]
+      k <- 1
+    }else{
+      k = k + 1
+    }
+  }
+  countObs <- c(countObs, k)
+  names <- c(names, sid[length(t)])
+  names(countObs) <- names
+  
+  # Do not work anymore without sid field
+  # sid <- dataBaseFiltered[,fields["sid"]]
+  # countObs <- table(sid) # count of observation by storm
+  
+  row <- max(countObs) # maximum number of rows
+  len <- length(unique(sid)) # number of unique storm in csv
+  cumulativeIndex <- cumsum(countObs) # starting points for each storm in csv
+  
+  # Initialize template structure
+  templateArray = array(NaN, dim=c(row,len))
+  
+  # Mandatory fields
+  data <- list(
+    names = array(NaN, dim=len),
+    seasons = array(NaN, dim=len),
+    isotimes = templateArray,
+    longitude = templateArray,
+    latitude = templateArray,
+    msw = templateArray
+  )
+  
+  # TODO Remove later
+  if("sshs" %in% names(fields)){
+    data$sshs <- templateArray
+  }
+  
+  if("rmw" %in% names(fields)){
+    data$rmw <- templateArray
+  }
+  
+  if("pressure" %in% names(fields)){
+    data$pressure <- templateArray
+  }
+  
+  if("poci" %in% names(fields)){
+    data$poci <- templateArray
+  }
+  
+  for(i in 1:len){
+    
+    # Get rows observations boundaries
+    if(i!=1){
+      start = cumulativeIndex[i-1] + 1
+    }else{
+      start = 1
+    }
+    end = cumulativeIndex[i]
+    
+    # Fill data
+    data$names[i] <- dataBaseFiltered[start, fields["names"]]
+    data$seasons[i] <- as.numeric(dataBaseFiltered[start, fields["seasons"]])
+    data$isotimes[,i] <- c(dataBaseFiltered[start:end, fields["isoTime"]], rep(NaN, row-countObs[i]))
+    data$longitude[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["lon"]], rep(NaN, row-countObs[i])))
+    data$latitude[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["lat"]], rep(NaN, row-countObs[i])))
+    
+    
+    data$msw[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["msw"]], rep(NaN, row-countObs[i])))
+    
+    
+    # TODO Remove later
+    if ("sshs" %in% names(fields)) {
+      data$sshs[,i] <- as.numeric(c(dataBaseFiltered[start:end, fields["sshs"]], rep(NaN, row-countObs[i])))
+    }
+    
```

Comment:
TBD

Reply:
I meant that it will be removed with the merge of PR 40